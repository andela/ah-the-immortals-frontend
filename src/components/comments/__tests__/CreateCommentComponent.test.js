import React from 'react';
import { Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { createBrowserHistory } from 'history';
import CreateComment from '../CreateCommentComponent';

describe('Tests CreateCommentsComponent', () => {
  const history = createBrowserHistory();
  const props = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
    value: ''
  };
  const wrapper = mount(
    <Router history={history}>
      <CreateComment {...props} />
    </Router>
  );
  it('tests mounting of component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
