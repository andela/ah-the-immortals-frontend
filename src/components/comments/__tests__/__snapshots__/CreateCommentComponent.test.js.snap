// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests CreateCommentsComponent tests mounting of component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <CreateComment
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
    value=""
  >
    <div
      className="mb-5"
    >
      <h3>
        Leave your comment
      </h3>
      <form
        className="form-horizontal"
        onSubmit={[MockFunction]}
      >
        <div
          className="form-group"
        >
          <textarea
            className="form-control"
            name="body"
            onChange={[MockFunction]}
            placeholder="Leave your thoughts on this article here"
            required={true}
            rows="6"
            value=""
          />
        </div>
        <div
          className="form-item"
        >
          <div
            className=""
          >
            <Button
              active={false}
              disabled={false}
              type="submit"
              variant="primary"
            >
              <button
                className="btn btn-primary"
                disabled={false}
                type="submit"
              >
                ADD COMMENT
              </button>
            </Button>
          </div>
        </div>
      </form>
      <hr />
    </div>
  </CreateComment>
</Router>
`;
